src/Forms/Shared/Samples/Data/CreateMobileGeodatabase/CreateMobileGeodatabase.xaml.cs

```diff
@@ -0,0 +1,219 @@
+        }
+
+        private async Task Initialize()
+        {
+            // Create a map.
+            MyMapView.Map = new Map(BasemapStyle.ArcGISTopographic);
+            MyMapView.SetViewpoint(new Viewpoint(39.323845, -77.733201, 10000.0));
+
+            await CreateGeodatabase();
+        }
+
+        private async Task CreateGeodatabase()
+        {
+            try
+            {
+                // Create a new randomly named directory for the geodatabase.
+#if WINDOWS_UWP
+                _directoryPath = Path.Combine(Environment.ExpandEnvironmentVariables("%TEMP%"), "CreateMobileGeodatabase", Guid.NewGuid().ToString());
+#elif __IOS__ || __ANDROID__
+                _directoryPath = Path.Combine(Environment.GetFolderPath(Environment.SpecialFolder.MyDocuments), "CreateMobileGeodatabase", Guid.NewGuid().ToString());
+#endif
+
+                if (!Directory.Exists(_directoryPath))
+                {
+                    Directory.CreateDirectory(_directoryPath);
+                }
+
+                // Create the geodatabase file.
+                _gdbPath = Path.Combine(_directoryPath, "LocationHistory.geodatabase");
+
+                // Delete existing file if present from previous sample run.
+                if (File.Exists(_gdbPath))
+                {
+                    File.Delete(_gdbPath);
+                }
+
+                _geodatabase = await Geodatabase.CreateAsync(_gdbPath);
+
+                // Construct a table description which stores features as points on map.
+                var tableDescription = new TableDescription("LocationHistory", SpatialReferences.Wgs84, GeometryType.Point)
+                {
+                    HasAttachments = false,
+                    HasM = false,
+                    HasZ = false
+                };
+
+                // Set up the fields for the table,
+                // FieldType.OID is the primary key of the SQLite table
+                // FieldType.Date is a date column used to store a Calendar date
+                // FieldDescriptions can be a SHORT, INTEGER, GUID, FLOAT, DOUBLE, DATE, TEXT, OID, GLOBALID, BLOB, GEOMETRY, RASTER, or XML.
+                tableDescription.FieldDescriptions.Add(new FieldDescription("oid", FieldType.OID));
+                tableDescription.FieldDescriptions.Add(new FieldDescription("collection_timestamp", FieldType.Date));
+
+                // Add a new table to the geodatabase by creating one from the table description.
+                _featureTable = await _geodatabase.CreateTableAsync(tableDescription);
+
+                // Clear the table.
+                await UpdateTable();
+
+                // Create a feature layer for the map.
+                FeatureLayer featureLayer = new FeatureLayer(_featureTable);
+                MyMapView.Map.OperationalLayers.Add(featureLayer);
+
+                // Add an event handler for new feature taps.
+                MyMapView.GeoViewTapped += MyMapView_GeoViewTapped;
```

Comment:
Could this be done in the xaml? Does it add more clarity here than it would in the xaml?

Reply:
I do it in code so that the user can't attempt to place before the geodatabase is created. This is useful when the user has closed a gdb but hasn't opened a new one yet.